components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/main/java/org/wso2/carbon/identity/api/server/action/management/v1/core/ServerActionManagementService.java

```diff
@@ -0,0 +1,276 @@
+
+                actionTypesResponseItems.add(new ActionTypesResponseItem()
+                        .type(ActionTypesResponseItem.TypeEnum.valueOf(actionType.getActionType()))
+                        .displayName(actionType.getDisplayName())
+                        .description(actionType.getDescription())
+                        .count(actionsCountPerType.get(actionType.getActionType()) == null ? 0 :
+                                actionsCountPerType.get(actionType.getActionType()))
+                        .self(ActionMgtEndpointUtil.buildURIForActionType(actionType.getActionType())));
+            }
+
+            return actionTypesResponseItems;
+        } catch (ActionMgtException e) {
+            throw ActionMgtEndpointUtil.handleActionMgtException(e);
+        }
+    }
+
+    /**
+     * Build ActionResponse from Action.
+     *
+     * @param action Action object.
+     * @return ActionResponse object.
+     */
+    private ActionResponse buildActionResponse(Action action) {
+
+        return new ActionResponse()
+                .id(action.getId())
+                .type(ActionResponse.TypeEnum.valueOf(action.getType().toString()))
+                .name(action.getName())
+                .description(action.getDescription())
+                .status(ActionResponse.StatusEnum.valueOf(action.getStatus().toString()))
+                .endpoint(new Endpoint()
+                        .uri(action.getEndpoint().getUri())
+                        .authentication(new AuthenticationType()
+                                .type(AuthenticationType.TypeEnum.valueOf(action.getEndpoint()
+                                        .getAuthentication().getType().toString()))
+                                .properties(action.getEndpoint().getAuthentication().getProperties())));
+    }
+
+    /**
+     * Create Action from the Action model.
+     *
+     * @param actionModel API model.
+     * @return Action.
+     */
+    private Action buildAction(ActionModel actionModel) {
+
+        ActionMgtEndpointUtil.validateAction(actionModel);
+        Action.ActionRequestBuilder actionRequestBuilder = new Action.ActionRequestBuilder()
+                .name(actionModel.getName())
+                .description(actionModel.getDescription())
+                .endpoint(new EndpointConfig.EndpointConfigBuilder()
+                        .uri(actionModel.getEndpoint().getUri())
+                        .authentication(new AuthType.AuthTypeBuilder()
+                                .type(AuthType.TypeEnum.valueOf(actionModel.getEndpoint().getAuthentication()
+                                        .getType().toString()))
+                                .properties(actionModel.getEndpoint().getAuthentication().getProperties())
+                                .build())
+                        .build());
+
+        return actionRequestBuilder.build();
+    }
+}
```

Comment:
Ideally we should have added audit logs for every add, update, delete action management operations in minimal

It's fine to improve this in next phase as well.
Audit logs should be ideally added from service layer (not rest api layer)

Reply:
Will add audit logs in a separate effort.

Reply:
Issue to track that task: https://github.com/wso2-enterprise/asgardeo-product/issues/25290